src/Tizen.Multimedia.Vision/MediaVision/InferenceFaceDetector.cs

```diff
@@ -0,0 +1,204 @@
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public async Task<InferenceFaceDetectorResult> InferenceAsync(MediaVisionSource source)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            InteropFD.InferenceAsync(_handle, source.Handle).Validate("Failed to inference face detection.");
+
+            return await Task.Factory.StartNew(() => new InferenceFaceDetectorResult(_handle),
+                CancellationToken.None,
+                TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning,
+                TaskScheduler.Default);
+        }
+
+        /// <summary>
+        /// Requests to detect face on the given source image.<br/>
+        /// </summary>
+        /// <remarks>
+        /// This API is not guranteed that inference is done when this method returns. The user can get the result by using <see cref="GetBoundBox"/>.<br/>
+        /// And the user call this API again before the previous one is finished internally, API call will be ignored until the previous one is finished.<br/>
+        /// If there's no detected face, <see cref="InferenceFaceDetectorResult.BoundBox"/> will be empty.<br/>
+        /// Note that this API could use about twice as much memory as <see cref="InferenceFaceDetector.Inference"/>.
+        /// </remarks>
+        /// <param name="source">The image data to detect face.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceDetector already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <seealso cref="GetBoundBox"/>
+        /// <since_tizen> 12 </since_tizen>
+        public void RequestInference(MediaVisionSource source)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            InteropFD.InferenceAsync(_handle, source.Handle).Validate("Failed to inference face detection.");
+        }
+
+        /// <summary>
+        /// Gets the bound box as a result of <see cref="RequestInference"/>.
+        /// </summary>
+        /// <remarks>
+        /// If there's no detected face, <see cref="InferenceFaceDetectorResult.BoundBox"/> will be empty.<br/>
+        /// This API uses about twice as much memory as <see cref="InferenceFaceDetector.Inference"/>.
+        /// </remarks>
+        /// <returns>A bound box of detected face.</returns>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceDetector already has been disposed.</exception>
+        /// <seealso cref="RequestInference"/>
+        /// <since_tizen> 12 </since_tizen>
+        public InferenceFaceDetectorResult GetBoundBox()
```

Comment:
If think those methods (GetBoundBox(), GetLabel(), ...) can be named just "GetResult()"
What if there was not inference requested already? InvalidOperation exception would be fitting, but maybe empty result is returned?

Reply:
When deciding on a function name, it would be best to choose one that allows you to guess what result can be obtained from it. GetResult() has implicit mearning.

Reply:
I think it could be changed to `GetRequestResults()`, because it's include request id also and it could be added more property in the future.